python/packages/autogen-ext/src/autogen_ext/agents/web_surfer/_multimodal_web_surfer.py

```diff
@@ -59,67 +55,122 @@
 )
 from ._types import InteractiveRegion, UserContent
 from ._utils import message_content_to_str
+from .playwright_controller import PlaywrightController
+
 
-# Viewport dimensions
-VIEWPORT_HEIGHT = 900
-VIEWPORT_WIDTH = 1440
+class MultimodalWebSurfer(BaseChatAgent):
+    """
+    MultimodalWebSurfer is a multimodal agent that acts as a web surfer that can search the web and visit web pages.
 
-# Size of the image we send to the MLM
-# Current values represent a 0.85 scaling to fit within the GPT-4v short-edge constraints (768px)
-MLM_HEIGHT = 765
-MLM_WIDTH = 1224
+    It launches a chromium browser and allows the playwright to interact with the web browser and can perform a variety of actions. The browser is launched on the first call to the agent and is reused for subsequent calls.
 
-SCREENSHOT_TOKENS = 1105
+    It must be used with a multimodal model client that supports tool calling, ideally GPT-4o currently.
 
+    .. note::
+        Please note that using the MultimodalWebSurfer involves interacting with a digital world designed for humans, which carries inherent risks.
+        Be aware that agents may occasionally attempt risky actions, such as recruiting humans for help or accepting cookie agreements without human involvement. Always ensure agents are monitored and operate within a controlled environment to prevent unintended consequences.
+        Moreover, be cautious that MultimodalWebSurfer may be susceptible to prompt injection attacks from webpages.
+
+    Args:
+        name (str): The name of the agent.
+        model_client (ChatCompletionClient): The model client used by the agent.
```

Comment:
Add notes on the requirement on the model. 

And in the constructor validate model capabilities

Reply:
done